lib/metrics_target_resource.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
```

Comment:
Is there anything we need to be careful about persistence of this? We don't persist anything about this class (as opposed to our models), which is fine, we don't persist anything for `MonitorableResource` as well. Just wanted to be sure that restarts (which happens multiple times a day after each deployment) is OK from the perspective of how this operates.

Reply:
It should be okay, we should not miss out on any data if the restarts happen within ~30 mins.
We can have retransmit for the same data in some cases (eg. the data was written but not moved to `done`). In this case Victoriametrics will dedup samples within a configured `-dedup.minScrapeInterval`, which we can set set to match the scrape interval 15s.

Reply:
I just noticed that this parameter does not have a default, added it in #3267